be/src/storage/rowset/segment_v2/page_io.cpp

```diff
@@ -110,6 +113,56 @@ Status PageIO::write_page(fs::WritableBlock* wblock, const std::vector<Slice>& b
     return Status::OK();
 }
 
+Status PageIO::decompress_bitshuffle_page(const PageReadOptions& opts, Slice& page_slice, PageFooterPB* footer,
+                                          uint32_t footer_size, std::unique_ptr<char[]>& page) {
+    size_t dict_header_size = BINARY_DICT_PAGE_HEADER_SIZE;
+    size_t bitshuffle_header_size = BITSHUFFLE_PAGE_HEADER_SIZE;
+    size_t type = EncodingTypePB::BIT_SHUFFLE;
+    if (opts.encoding_type == EncodingTypePB::DICT_ENCODING) {
+        type = decode_fixed32_le((const uint8_t*)&page_slice.data[0]);
```

Comment:
I don't think it's a good design to expose the encoding details of the dictionary page to the PageIO module.